`ultralytics 8.3.247` Improve Ray Tune trial names logged to W&B

raimbekovm opened the pull request:
Fixes #23000

When using `model.tune(use_ray=True)`, all trials appeared in WandB with the default name 'train' instead of unique trial-specific names.

**Root cause:** `train_args.pop('name')` removed the base name before trials were created, so `config['name']` was never set.

**Solution:**
- Save `base_name` before it's popped from `train_args`
- Use `tune.get_trial_id()` to generate unique trial-specific names
- Format: `{base_name}_{trial_suffix}` (e.g., 'my_run_00000')

**Result:**
WandB run names now align with local directory structure:
- Local: `my_run/_tune_2c2fc_00000/`
- WandB: `my_run_00000`

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Fix Ray Tune hyperparameter tuning so each W&B run gets a unique, trial-specific name 🎛️📝

### 📊 Key Changes
- Adds Ray Tune `trial_id`-based suffixing to `config["name"]` inside `_tune()` so W&B logs don’t collide across trials
- Preserves a `base_name` from `train_args` before training args are consumed, then uses it for consistent trial naming
- Includes a safe fallback to the base name when not running under Ray Tune (or if trial ID retrieval fails)

### 🎯 Purpose & Impact
- Prevents W&B runs from overwriting/merging due to identical names across Ray Tune trials ✅
- Makes trial results easier to track and compare in W&B during tuning 🔎
- Improves reproducibility and debugging by keeping per-trial logging clearly separated 📌

raimbekovm committed:
Fix WandB trial names in Ray Tune hyperparameter tuning

Fixes #23000

When using model.tune(use_ray=True), all trials were appearing in WandB
with the default name 'train' instead of unique trial-specific names.

Root cause: train_args.pop('name') removed the base name before trials
were created, so config['name'] was never set for individual trials.

Solution:
- Save base_name before it's popped from train_args
- Use tune.get_trial_id() to generate unique trial-specific names
- Format: {base_name}_{trial_suffix} (e.g., 'my_run_00000')
- Fallback to base_name if not in Ray Tune context

This aligns WandB run names with local directory structure:
  Local: my_run/_tune_2c2fc_00000/
  WandB: my_run_00000

UltralyticsAssistant commented:
👋 Hello @raimbekovm, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with next steps—an engineer will also review and assist shortly 🛠️🙂

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

UltralyticsAssistant reviewed (commented):
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Good intent (unique W&B names per Ray Tune trial), but the current ordering introduces a runtime error: `base_name` is used before assignment. Move the `base_name` capture earlier in `_tune()` (before the trial-name suffix logic) to make this safe.

💬 Posted 2 inline comments

UltralyticsAssistant commented on ultralytics/utils/tuner.py:
```diff
@@ -93,6 +93,16 @@ def _tune(config):
+        try:
+            trial_id = tune.get_trial_id()  # Get current trial ID (e.g., "2c2fc_00000")
+            trial_suffix = trial_id.split("_")[-1] if "_" in trial_id else trial_id
+            config["name"] = f"{base_name}_{trial_suffix}"
```
❗ **CRITICAL**: `base_name` is referenced here before it’s assigned in this function. In the diff, `base_name` is defined later (R 137), so this will raise `UnboundLocalError` the first time this code path runs. Move the `base_name = ...` assignment to a point *before* this block (ideally before any mutation/consumption of `train_args`).

raimbekovm committed:
Fix UnboundLocalError: move base_name before _tune() definition

Move base_name assignment before the _tune() function definition to fix
closure scope issue. Previously base_name was used inside _tune() but
defined after the function, causing UnboundLocalError at runtime.

glenn-jocher committed:
Merge branch 'main' into fix-wandb-raytune-naming

glenn-jocher committed:
Update __init__.py

Signed-off-by: Glenn Jocher <glenn.jocher@ultralytics.com>

glenn-jocher committed:
Merge branch 'main' into fix-wandb-raytune-naming
